SlevomatCodingStandard/Helpers/SniffLocalCache.php

```diff
@@ -0,0 +1,100 @@
+<?php declare(strict_types = 1);
+
+namespace SlevomatCodingStandard\Helpers;
+
+use Closure;
+use PHP_CodeSniffer\Files\File;
+use function array_key_exists;
+
+final class SniffLocalCache
+{
+
+	/**
+	 * @phpcsSuppress SlevomatCodingStandard.TypeHints.DisallowMixedTypeHint
+	 * @var array<string, mixed>
+	 */
+	private $cache = [];
+
+	public function has(File $phpcsFile): bool
+	{
+		$key = $this->key($phpcsFile);
+
+		return array_key_exists($key, $this->cache);
+	}
+
+	/**
+	 * @phpcsSuppress SlevomatCodingStandard.TypeHints.DisallowMixedTypeHint
+	 * @param File $phpcsFile
+	 * @return mixed
+	 */
+	public function get(File $phpcsFile)
+	{
+		return $this->cache[$this->key($phpcsFile)] ?? null;
+	}
+
+	/**
+	 * @phpcsSuppress SlevomatCodingStandard.TypeHints.DisallowMixedTypeHint
+	 * @param File $phpcsFile
+	 * @param mixed $value
+	 */
+	public function set(File $phpcsFile, $value): void
+	{
+		$key = $this->key($phpcsFile);
+		$this->cache[$key] = $value;
+		$this->removeOldCache($phpcsFile);
+	}
+
+	private function setIfNotCached(File $phpcsFile, Closure $lazyValue): void
+	{
+		if ($this->has($phpcsFile)) {
+			return;
+		}
+
+		$this->set($phpcsFile, $lazyValue());
+	}
+
+	/**
+	 * @phpcsSuppress SlevomatCodingStandard.TypeHints.DisallowMixedTypeHint
+	 * @param File    $phpcsFile
+	 * @param Closure $lazyValue
+	 * @return mixed
+	 */
+	public function getAndSetIfNotCached(File $phpcsFile, Closure $lazyValue)
```

Comment:
Please add `$key` parameter so we can use `SniffLocalCache` later for use statements as well.

Reply:
it is not global or static, just `new SniffLocalCache()` inside use statements sniff and it will save use statements cache inside its helper method or maybe i misunderstand you?

Reply:
Oh, you are right.